### ISSUE TITLE
`check-multi-line-strings` option to only check first and maybe last line

### ISSUE THREAD

**Issue by MaPePeR** [adrienverge/yamllint#517]
If you use some sort of code blocks in multi line strings - for example bash or jinja2 templates - you might want to indent the code blocks properly. Currently the `check-multi-line-strings`, when set to `true`, reports errors if you do that:

```yaml
---
code: |
  if []; then
    echo "Do something"
  fi
```

with `.yamllint`:
```yaml
---
rules:
  indentation:
    spaces: 2
    check-multi-line-strings: true
```

reports

```
  4:5       error    wrong indentation: expected 2 but found 4  (indentation)
```

I think there is merit in having a third (or maybe forth) option for `check-multi-line-strings`: 
* `first` only check indentation of first line
* `border` check first and last line of multi line string for indentation (better name suggestions appreciated. I don't really like it, but can't come up with a better word at the moment)

Because just setting the option to `false` does not report any errors if I do this (Multi line string indented with 8 spaces instead of 2):
```yaml
---
code: |
        if []; then
          echo "Do something"
        fi
```

**Comment by adrienverge** [comment]
Hello, thanks for your feedback!

This is an interesting idea. These options could also be named:
```yaml
check-multi-line-strings: false | true | first-line-only | first-and-last-lines
```

I'm not sure whether the last option (`first-and-last-lines`) really makes sense, because it would report errors on valid indentations like:
```yaml
code: |
  for i in range(10):
    print(i)
```

**Comment by MaPePeR** [comment]
Your name suggestions are a lot better than mine.

I'm also unsure about the usefulness of `first-and-last-lines`. Its probably very situational. For bash and jinja2 templates its probably fine - for python code it isn't. I thought of it as an intermediate that is more strict than just `first-line-only` and closer to the `true` option, but still allowing indentation inside the multi line "block" (So the last line is indented like a "closing brace" for the multi line string, but since yaml doesn't have "closing braces" that's probably not expected anyway).
